include/SyntaxTree.h

```diff
@@ -30,23 +34,81 @@ class SyntaxTree {
     SyntaxTree(const SyntaxTree&) = delete;
     SyntaxTree& operator=(const SyntaxTree&) = delete;
 
-    /// Creates a syntax tree from file on disk or text in memory.
-    static SyntaxTree fromFile(SourceManager& sourceManager, StringRef path);
-    static SyntaxTree fromText(SourceManager& sourceManager, StringRef text);
+    /// Creates a specialized syntax tree from text in memory.
+    template<typename T>
+    static SyntaxTree fromText(StringRef text) {
+        return fromText<T>(text, getDefaultSourceManager());
+    }
+
+    template<typename T>
+    static SyntaxTree fromText(StringRef text, SourceManager& sourceManager) {
+        return create<T>(sourceManager, sourceManager.assignText(text));
+    }
+
+    /// Creates a syntax tree from a full compilation unit.
+    static SyntaxTree fromFile(StringRef path) {
+        return fromFile(path, getDefaultSourceManager());
+    }
 
-    /// Gets the root of the syntax tree.
-    const CompilationUnitSyntax* root() const { return rootNode; }
+    static SyntaxTree fromText(StringRef text) {
+        return fromText(text, getDefaultSourceManager());
+    }
+
+    static SyntaxTree fromFile(StringRef path, SourceManager& sourceManager) {
```

Comment:
No need to heap allocate if it's not necessary; I'm a big fan of move semantics.